node_helper.js

```diff
@@ -15,15 +15,34 @@ module.exports = NodeHelper.create({
 		if(notification === "GET_COMIC"){
 			
 			var comicJsonUri = payload.config.dailyJsonUrl;
+			var comic;
+			var rndcomic;
+			var rndUrl;
+			var body;
+			var d = new Date();
+			var n = d.getDay(); 
 			
 			request(comicJsonUri, function (error, response, body) {
 				if (!error && response.statusCode == 200) {
 					console.log(body);
-					self.sendSocketNotification("COMIC", JSON.parse(body));
-					console.log(JSON.parse(body).img);
-					
+					if ( this.n == 1 || this.n == 3 || this.n == 5 ) {
```

Comment:
Well i did say it was a bit of a hack, but yes it does work. Has been running on my mirror for quite a while now.. I'm not at all versed in node/js so my process exists of trying things and when it works,,,,  it works :)

I'll go ahead and remove any unneeded vars..




Date: Wed, 10 Aug 2016 08:05:33 -0700
From: notifications@github.com
To: DailyXKCD@noreply.github.com
CC: ocp_student@hotmail.com; author@noreply.github.com
Subject: Re: [Blastitt/DailyXKCD] Added getting random conmic on days when there isn't a new one. Also … (#1)

In node_helper.js:

>  			
>  			request(comicJsonUri, function (error, response, body) {
>  				if (!error && response.statusCode == 200) {
>  					console.log(body);
> -					self.sendSocketNotification("COMIC", JSON.parse(body));
> -					console.log(JSON.parse(body).img);
> -					
> +					if ( this.n == 1 || this.n == 3 || this.n == 5 ) {

Have you tested this? Since this is executed within a callback function, I don't think this.n is referring to the local var n declared earlier. Same for the other references to "this" later in the function body.


—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.